cmd/kubeadm/app/cmd/phases/preflight.go

```diff
@@ -46,31 +51,54 @@ var (
 		# Run node pre-flight checks.
 		kubeadm alpha phase preflight node
 	`)
+
+	errorMissingConfigFlag = errors.New("the --config flag is mandatory")
 )
 
 // NewCmdPreFlight calls cobra.Command for preflight checks
 func NewCmdPreFlight() *cobra.Command {
+	var cfgPath string
+	var ignorePreflightErrors []string
+
 	cmd := &cobra.Command{
 		Use:   "preflight",
 		Short: "Run pre-flight checks",
 		Long:  cmdutil.MacroCommandLongDescription,
 	}
 
-	cmd.AddCommand(NewCmdPreFlightMaster())
-	cmd.AddCommand(NewCmdPreFlightNode())
+	options.AddConfigFlag(cmd.PersistentFlags(), &cfgPath)
+	options.AddIgnorePreflightErrorsFlag(cmd.PersistentFlags(), &ignorePreflightErrors)
+
+	cmd.AddCommand(NewCmdPreFlightMaster(&cfgPath, &ignorePreflightErrors))
+	cmd.AddCommand(NewCmdPreFlightNode(&cfgPath, &ignorePreflightErrors))
+
 	return cmd
 }
 
 // NewCmdPreFlightMaster calls cobra.Command for master preflight checks
-func NewCmdPreFlightMaster() *cobra.Command {
+func NewCmdPreFlightMaster(cfgPath *string, ignorePreflightErrors *[]string) *cobra.Command {
+
 	cmd := &cobra.Command{
 		Use:     "master",
 		Short:   "Run master pre-flight checks",
 		Long:    masterPreflightLongDesc,
 		Example: masterPreflightExample,
 		Run: func(cmd *cobra.Command, args []string) {
-			cfg := &kubeadmapi.InitConfiguration{}
-			err := preflight.RunInitMasterChecks(utilsexec.New(), cfg, sets.NewString())
+			if len(*cfgPath) == 0 {
+				kubeadmutil.CheckErr(errorMissingConfigFlag)
+			}
+			ignorePreflightErrorsSet, err := validation.ValidateIgnorePreflightErrors(*ignorePreflightErrors)
+			kubeadmutil.CheckErr(err)
+
+			cfg := &kubeadmapiv1alpha3.InitConfiguration{}
+			kubeadmscheme.Scheme.Default(cfg)
+
+			internalcfg, err := configutil.ConfigFileAndDefaultsToInternalConfig(*cfgPath, cfg)
+			kubeadmutil.CheckErr(err)
+			err = configutil.VerifyAPIServerBindAddress(internalcfg.APIEndpoint.AdvertiseAddress)
+			kubeadmutil.CheckErr(err)
+
+			err = preflight.RunInitMasterChecks(utilsexec.New(), internalcfg, ignorePreflightErrorsSet)
```

Comment:
@neolit123 not blocking, but could you kindly add `fmt.Println("[preflight] running pre-flight checks")` + eventually a `pre-flight checks passed` after the error check?
This will provide a better UX (now the user doesn't gets any message in case all the checks are passing).
same for nodes